projects/aca-shared/src/lib/components/tool-bar/toolbar-menu/toolbar-menu.component.html

```diff
@@ -15,11 +15,18 @@
   <ng-container *ngFor="let child of actionRef.children; trackBy: trackByActionId">
     <ng-container [ngSwitch]="child.type">
       <ng-container *ngSwitchCase="'custom'">
-        <adf-dynamic-component [id]="child.component" [data]="child.data"></adf-dynamic-component>
+        <button mat-menu-item class="aca-mat-menu-item-container" (click)="onCustomItemContainerClick($event)">
```

Comment:
not sure this is correct, the dynamic component can be anything, and we should already have cases when it's a button (like toggle favorite)... so wrapping button with a button does not make sense

Reply:
I agree with the 'button in a button' aspect of this approach. However, the reason why we took this approach was due to how angular material handles menu items. The mat-menu-item directive is responsible for handling a lot of functionality associated with the menu items. This also includes features such as keyboard navigation and interactions. However, according to some of the comments made by some developers from the @angular/components repository as well as logged issues (links [here](https://github.com/angular/components/issues/25323#issuecomment-1222093609) and [here](https://github.com/angular/components/issues/18123), unfortunately I couldn't find any official 'documentation' for this point), this directive only works when it is a direct child of a mat-menu component. It WILL NOT WORK if the mat-menu item is added to the mat-menu via content-projection (for e.g. with custom components such as  adf-dynamic-components as well as adf-toolbar-menu-item). Right now, the buttons inside adf-dynamic-component and adf-toolbar-menu-item are the ones that have the mat-menu-item directive.
The reason why it seems to currently work with adf-toolbar-menu-items is due to a workaround implemented in that component where we manually push the menu items into the mat-menu (see [ngAfterViewInit()](https://github.com/Alfresco/alfresco-content-app/blob/develop/projects/aca-shared/src/lib/components/tool-bar/toolbar-menu/toolbar-menu.component.ts#L59) implementation). I think that this kind of approach might not be possible with the adf-dynamic-components because it can be designed in any manner outside of our control, by third party developers using our applications. Restricting them to follow a certain kind of component architecture, so that we can make changes similar to adf-toolbar-menu-items is something that can be done, but might not be desirable.
As for why we used a button, we could use a div/span with a custom click handler and keydown.enter/keydown.space events. However, there might be some accessibility issues related to using them (for e.g. those related to  focus and screen reader related). We can investigate into this aspect a bit more if needed though.